griptape/schemas/base_schema.py

```diff
@@ -1,5 +1,79 @@
+from __future__ import annotations
+from abc import ABC
+from typing import Union, get_origin, get_args, Any
+import attrs
 from marshmallow import Schema, fields
 
+from griptape.schemas.bytes_field import Bytes
+
 
 class BaseSchema(Schema):
-    schema_namespace = fields.Str(allow_none=True)
+    DATACLASS_TYPE_MAPPING = {**Schema.TYPE_MAPPING, dict: fields.Dict, bytes: Bytes}
+
+    @classmethod
+    def from_attrs_cls(cls, attrscls):
+        """Generate a Schema from an attrs class.
+
+        Args:
+            attrscls: An attrs class.
+        """
+
+        from marshmallow import post_load
+        from griptape.utils import PromptStack
+        from griptape.structures import Structure
+        from griptape.drivers import BaseConversationMemoryDriver, BasePromptDriver
+
+        class SubSchema(cls):
+            @post_load
+            def make_obj(self, data, **kwargs):
+                return attrscls(**data)
+
+        attrs.resolve_types(
+            attrscls,
+            localns={
+                "PromptStack": PromptStack,
+                "Input": PromptStack.Input,
+                "Structure": Structure,
+                "BaseConversationMemoryDriver": BaseConversationMemoryDriver,
+                "BasePromptDriver": BasePromptDriver,
+            },
+        )
```

Comment:
Semi arbitrary feeling imports to make this work. Without these we get a `NameError` on `resolve_types`. I think it has to do with import order.